muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/coursepicker/CoursePickerRESTService.java

```diff
@@ -407,7 +407,20 @@ public Response listWorkspaces(
                 String educationTypeId = (String) result.get("educationTypeIdentifier");
                 String educationTypeName = (String) result.get("educationTypeName");
                 SchoolDataIdentifier educationSubtypeId = SchoolDataIdentifier.fromId((String) result.get("educationSubtypeIdentifier"));
-
+                SchoolDataIdentifier schoolDataIdentifier = SchoolDataIdentifier.fromId((String) result.get("identifier"));
```

Comment:
`IndexedWorkspace` (jota käytetään työtilan indeksoinnissa), käyttää sisäisesti `id` ja `identifier`-kenttien arvona samaa tietoa, eli tässä metodissa jo aiemmin luettu `workspaceIdentifier` on käytännössä aina sama kuin tämä, eli alla oleva tsekki näiden yhteneväisyydestä on luultavasti turha ellen ihan väärin tsekannut.

Sivuhuomiona, koska `schoolDataIdentifier` ja `workspaceIdentifier` eivät alla olevan loopin sisällä muutu, niin kyseinen `if` voisi olla loopin ulkopuolella, jolloin loop ja sen tarvitsemat muuttujat (subjectList) voidaan skipata kokonaan, jos eivät mätsää. Olisi näin hiuksenhienosti tehokkaampi koodi, mutta tosiaan tuo tsekki on varmaankin turha.